File: frontend/src/types/question.ts
import type { Tag } from './tag'

export type QuestionType =
  | 'TRANSLATION_ZH_TO_JA'
  | 'TRANSLATION_ZH_TO_JA_ARTICLE'
  | 'TRANSLATION_EN_TO_JA'
  | 'TRANSLATION_EN_TO_JA_ARTICLE'

export type QuestionAnswer = {
  id: number
  answerText: string
  answerType: 'STANDARD' | 'REFERENCE'
  primaryAnswer: boolean
  sortOrder: number
}

export type Question = {
  id: number
  questionType: QuestionType
  sourceText: string
  contextText: string
  level: string
  difficulty: number
  grammarPoint: string
  spoken: boolean
  business: boolean
  exam: boolean
  sourceType: 'AI' | 'MANUAL' | 'REVIEW_DERIVED'
  enabled: boolean
  tags: Tag[]
  answers: QuestionAnswer[]
  createdAt: string
  updatedAt: string
}

export type QuestionFormMode = 'create' | 'edit'

export type QuestionFormState = {
  sourceText: string
  contextText: string
  level: string
  difficulty: string
  grammarPoint: string
  spoken: boolean
  business: boolean
  exam: boolean
  tagCodes: string
  standardAnswer: string
  referenceAnswers: string
}

export type QuestionFilterState = {
  questionType: QuestionType
  level: string
  difficulty: string
  tagCodes: string
  sourceType: '' | 'AI' | 'MANUAL' | 'REVIEW_DERIVED'
  enabled: 'true' | 'false' | 'all'
  page: number
  size: number
}

export type RandomQuestionFilter = {
  questionType: QuestionType
  level: string
  difficulty: string
  tagCodes: string[]
}

export type AiQuestionGenerationPayload = {
  learningMode?: 'ZH_TO_JA' | 'EN_TO_JA'
  questionCount?: number
  level?: string
  difficulty?: number
  sceneTagCodes?: string[]
  functionTagCodes?: string[]
  excludedSourceTexts?: string[]
  extraRequirements?: string
}

export type AiArticleLengthTier = 'SHORT' | 'MEDIUM' | 'LONG'

export type AiArticleGenerationPayload = {
  learningMode?: 'ZH_TO_JA' | 'EN_TO_JA'
  level?: string
  difficulty?: number
  lengthTier?: AiArticleLengthTier
  genreTagCode?: string
  topic?: string
  extraRequirements?: string
}

export type QuestionPayload = {
  questionType: QuestionType
  sourceText: string
  contextText: string
  level: string
  difficulty: number
  grammarPoint: string
  spoken: boolean
  business: boolean
  exam: boolean
  tagCodes: string[]
  answers: {
    answerText: string
    answerType: 'STANDARD' | 'REFERENCE'
    primaryAnswer: boolean
    sortOrder: number
  }[]
}

File: backend/src/main/java/com/sazare/entity/Question.java
package com.sazare.entity;

import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
public class Question {

    /**
     * 主键ID
     */
    private Long id;

    /**
     * 题目类型：TRANSLATION_ZH_TO_JA=中译日短句，TRANSLATION_ZH_TO_JA_ARTICLE=中译日文章
     */
    private String questionType;

    /**
     * 中文题目原文，文章题按句使用双换行分隔
     */
    private String sourceText;

    /**
     * 题目语境说明
     */
    private String contextText;

    /**
     * JLPT等级：N5、N4、N3、N2、N1
     */
    private String level;

    /**
     * 难度等级，范围1到5
     */
    private Integer difficulty;

    /**
     * 短句语法点或文章生词提示
     */
    private String grammarPoint;

    /**
     * 是否口语表达
     */
    private Boolean spoken;

    /**
     * 是否商务表达
     */
    private Boolean business;

    /**
     * 是否考试相关
     */
    private Boolean exam;

    /**
     * 题目来源：AI=AI生成，MANUAL=人工录入，REVIEW_DERIVED=复习衍生题
     */
    private String sourceType;

    /**
     * 是否启用
     */
    private Boolean enabled;

    /**
     * 是否删除
     */
    private Boolean deleted;

    /**
     * 创建时间
     */
    private LocalDateTime createdAt;

    /**
     * 更新时间
     */
    private LocalDateTime updatedAt;
}
